File: iOS6bookExamples/ch22p651popovers/p550p565popovers/ExtraViewController.m
//
//  ExtraViewController.m
//  p550p565popovers
//
//  Created by Matt Neuburg on 1/22/13.
//
//

#import "ExtraViewController.h"

@interface ExtraViewController ()

@end

@implementation ExtraViewController
- (IBAction)doButton:(id)sender {
    NSLog(@"%@", NSStringFromCGRect(self.view.frame));
    [self dismissViewControllerAnimated:YES completion:^{
    }];
}

- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        // Custom initialization
        self.modalPresentationStyle = UIModalPresentationCurrentContext;
        self.contentSizeForViewInPopover = CGSizeMake(320,220);
        // self.modalTransitionStyle = UIModalTransitionStyleFlipHorizontal;
    }
    return self;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    // Do any additional setup after loading the view from its nib.
    self.view.frame = CGRectMake(0,0,320,220);
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

-(void)dealloc {
    NSLog(@"%@", @"dealloc");
}

@end

File: iOS9bookExamples/bk2ch09p476popovers/ch22p751popovers/ExtraViewController.swift


import UIKit

// summoned by "i" button in first popover
// notice that preferred content size is obeyed

class ExtraViewController : UIViewController {
    
    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: NSBundle?) {
        super.init(nibName: "ExtraViewController", bundle: nibBundleOrNil)
        self.modalPresentationStyle = .CurrentContext
        self.preferredContentSize = CGSizeMake(320,220)
    }
    
    required init(coder: NSCoder) {
        fatalError("NSCoding not supported")
    }
    
    @IBAction func doButton(sender:AnyObject) {
        print("extra view controller view frame: \(self.view.frame)")
        self.dismissViewControllerAnimated(true, completion:{
            print("dismissed")
            })
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.view.frame = CGRectMake(0,0,320,220)
    }
    
    deinit {
        print("dealloc extra view controller")
    }
    
}